src/dlgConnectionProfiles.cpp

```diff
@@ -62,21 +62,21 @@ dlgConnectionProfiles::dlgConnectionProfiles(QWidget * parent)
     connect_button = dialog_buttonbox->addButton(tr("Connect"), QDialogButtonBox::AcceptRole);
     connect_button->setIcon(QIcon(QStringLiteral(":/icons/dialog-ok-apply.png")));
 
-    connect(connect_button, SIGNAL(clicked()), this, SLOT(accept()));
-    connect(abort, SIGNAL(clicked()), this, SLOT(slot_cancel()));
-    connect(new_profile_button, SIGNAL(clicked()), this, SLOT(slot_addProfile()));
-    connect(copy_profile_button, SIGNAL(clicked()), this, SLOT(slot_copy_profile()));
-    connect(remove_profile_button, SIGNAL(clicked()), this, SLOT(slot_deleteProfile()));
-    connect(profile_name_entry, SIGNAL(textEdited(const QString)), this, SLOT(slot_update_name(const QString)));
-    connect(profile_name_entry, SIGNAL(editingFinished()), this, SLOT(slot_save_name()));
-    connect(host_name_entry, SIGNAL(textChanged(const QString)), this, SLOT(slot_update_url(const QString)));
-    connect(port_entry, SIGNAL(textChanged(const QString)), this, SLOT(slot_update_port(const QString)));
-    connect(autologin_checkBox, SIGNAL(stateChanged(int)), this, SLOT(slot_update_autologin(int)));
-    connect(login_entry, SIGNAL(textEdited(const QString)), this, SLOT(slot_update_login(const QString)));
-    connect(character_password_entry, SIGNAL(textEdited(const QString)), this, SLOT(slot_update_pass(const QString)));
-    connect(mud_description_textedit, SIGNAL(textChanged()), this, SLOT(slot_update_description()));
-    connect(profiles_tree_widget, SIGNAL(currentItemChanged(QListWidgetItem*, QListWidgetItem*)), this, SLOT(slot_item_clicked(QListWidgetItem*)));
-    connect(profiles_tree_widget, SIGNAL(itemDoubleClicked(QListWidgetItem*)), this, SLOT(accept()));
+    connect(connect_button, &QAbstractButton::clicked, this, &dlgConnectionProfiles::accept);
+    connect(abort, &QAbstractButton::clicked, this, &dlgConnectionProfiles::slot_cancel);
+    connect(new_profile_button, &QAbstractButton::clicked, this, &dlgConnectionProfiles::slot_addProfile);
+    connect(copy_profile_button, &QAbstractButton::clicked, this, &dlgConnectionProfiles::slot_copy_profile);
+    connect(remove_profile_button, &QAbstractButton::clicked, this, &dlgConnectionProfiles::slot_deleteProfile);
+    connect(profile_name_entry, &QLineEdit::textEdited, this, &dlgConnectionProfiles::slot_update_name);
```

Comment:
Does this *new* form of `connect(`...`)` handle the argument passing of some SIGNAL/SLOT things?

Reply:
Not new by a stretch - Qt5 was released in 2012, this is 6 years old. We're just really behind on adopting it. Yep it does

Reply:
Okay, I'll try and get used to using the ~*weird and unnatural*~ new-fangled form. 😝